src/main/java/bio/terra/service/dataset/IngestRequestValidator.java

```diff
@@ -30,6 +32,32 @@ public void validate(@NotNull Object target, Errors errors) {
     if (target instanceof IngestRequestModel) {
       IngestRequestModel ingestRequest = (IngestRequestModel) target;
       validateTableName(ingestRequest.getTable(), errors);
+
+      if (StringUtils.isEmpty(ingestRequest.getPath())
+          && !ingestRequest.getFormat().equals(IngestRequestModel.FormatEnum.ARRAY)) {
```

Comment:
could extract this variable out to `boolean isAPIIngest = ingestRequest.getFormat().equals(IngestRequestModel.FormatEnum.ARRAY)` for readability

Reply:
fair enough